### PR TITLE
Fixes #1660

### LINKED ISSUE #1660: Add field type protection to import-roaring
### Description

Currently import-roaring merges the imported data to the given field regardless of field type. This could easily corrupt data in mutex fields, int fields, and other field types which have restrictions on how the data can look. We need to disallow roaring-import into these fields or add protections which will reject the import if the data is corrupt.

### Success criteria (What criteria will consider this ticket closeable?)

It's impossible to create invalid data using the roaring-import endpoint. Invalid meaning a mutex field which has more than one bit set in the same column, or an int field with the not-null bit unset, but has integer data, or has integer data outside the configured range. This is not meant to be an exhaustive description of all ways data could be invalid.

### PR DESCRIPTION
**Pull request by yuce**
## Overview

Checks the field type on roaring import; if it's not a set field, returns an error.

Fixes #1660 

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [x] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: http/handler.go [hunk 1]
```diff
@@ -1473,11 +1473,16 @@ func (h *Handler) handlePostImportRoaring(w http.ResponseWriter, r *http.Request
 		return
 	}
 
+	resp := &pilosa.ImportResponse{}
 	// TODO give meaningful stats for import
 	err = h.api.ImportRoaring(r.Context(), urlVars["index"], urlVars["field"], shard, remote, body)
-	resp := &pilosa.ImportResponse{}
 	if err != nil {
 		resp.Err = err.Error()
+		if _, ok := err.(pilosa.BadRequestError); ok {
```
[the remark this answers is not in the document]
Comment (yuce) [reply http/handler.go]: I took it from `handleGetFragmentBlockData` about line: 1101 in `http/handler.go`

File: api.go [hunk 2]
```diff
@@ -321,13 +321,19 @@ func (api *API) ImportRoaring(ctx context.Context, indexName, fieldName string,
 	nodes := api.cluster.shardNodes(indexName, shard)
 	var eg errgroup.Group
 
+	field := api.holder.Field(indexName, fieldName)
+	if field == nil {
+		return newNotFoundError(ErrFieldNotFound)
+	}
+
+	// only set fields are supported
+	if field.Type() != "set" {
```
[the remark this answers is not in the document]
Comment (yuce) [reply api.go]: updated

File: http/handler.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (travisturner) [reply http/handler.go]: We should probably change that instance as well.
There are 14 instances of the `errors.Cause()`, and only 1 using `err.(Type)`, so it seems like the prevailing pattern is `errors.Cause()`.

[the remark this answers is not in the document]
Comment (travisturner) [reply http/handler.go]: Oh right, I think you have to switch on `type`, like:
```
switch errors.Cause(resp.Err).(type) {
    ...
}
```
But this is not critical, I was just pointing it out for consistency.

**Comment by yuce** [comment]
Is this good to be merged ?

**Pull request merged**